File: leetcode3/염혜정/3940. Limit Occurrences in Sorted Array.java
// o(n)

class Solution {
    public int[] limitOccurrences(int[] nums, int k) {
        int n = nums.length, idx = 0;
        int[] res = new int[n];
        int i = 0;
        while (i < n) {
            int j = i;
            while (j < n && nums[j] == nums[i]) j++;
            int count = Math.min(j - i, k);
            for (int c = 0; c < count; c++) res[idx++] = nums[i];
            i = j;
        }
        return java.util.Arrays.copyOf(res, idx);
    }
}

File: leetcode3/변지협/v2/3940. Limit Occurrences in Sorted Array.py

'''
1. 아이디어 :
2. 시간복잡도 :
    o(n * m)

3. 자료구조/알고리즘 :
'''
class Solution:
    def limitOccurrences(self, nums: list[int], k: int) -> list[int]:
        ans = []
        
        for i in nums:
            if len([j for j in ans if j == i]) == k:
                continue
            ans.append(i)
        
        return ans